### PR TITLE
Add in search filter with id

### PR DESCRIPTION
**Pull request by lamtu1**
Create a new searching way of using the `GeoName ID` to search for the place without typing in the name.

### CODE REVIEW

**Comment by lamtu1** [comment]
fixes #1443

**Comment by lamtu1** [comment]
`DEMO:` VIdeo showing the ID number input works on the GeoName location

https://github.com/osulp/Scholars-Archive/assets/111306182/b9255315-a697-49b8-a32f-5f1c6cc5fa85

**Comment by lamtu1** [comment]
Ready for review/approve and merge for QA.

File: .rubocop_fix_me.yml [hunk 1]
```diff
@@ -201,6 +201,7 @@ Style/RedundantParentheses:
     - 'lib/scholars_archive/validators/other_affiliation_validator.rb'
```
Comment (CGillen) [inline .rubocop_fix_me.yml:201]: Could you fix these? They shouldn't be too difficult, maybe even auto-solvable (`rubocop -A`)

**Review by CGillen — CHANGES_REQUESTED** [review]

File: lib/qa/authorities/geonames.rb [hunk 2]
```diff
@@ -47,6 +63,13 @@ def parse_authority_response(response)
       end
     end
 
+    # REFORMAT: Format data with the id given
+    def parse_response_with_id(response)
+      # Note: the trailing slash is meaningful.
+      [{ 'id' => "https://sws.geonames.org/#{response['geonameId']}/",
+        'label' => label.call(response, translate_fcl(response['fcl'])) }]
```
Comment (CGillen) [inline lib/qa/authorities/geonames.rb:70]: space here. that's `Layout/HashAlignment`

File: lib/qa/authorities/geonames.rb [hunk 3]
```diff
@@ -16,7 +16,23 @@ def search(q)
         Rails.logger.error 'Questioning Authority tried to call geonames, but no username was set'
         return []
       end
-      parse_authority_response(json(build_query_url(q)))
+
+      # CHECK: Making a check that if an id number or url is enter instead of city name
+      if q.to_s.match(/^[0-9]*$/)
+        parse_response_with_id(json(build_id_url(q)))
+      elsif q.to_s.match(%r{^https?:\/\/www\.geonames\.org})
```
Comment (CGillen) [inline lib/qa/authorities/geonames.rb:23]: I'm not sure what's different in OD2, but you don't need to escape the `//` part if the regex is in a `%r{}` block apparently
That's `Style/RedundantRegexpEscape`

**Review by CGillen — CHANGES_REQUESTED** [review]
`Metrics/MethodLength` is fine. It's a problem generally everywhere

**Comment by lamtu1** [comment]
@CGillen Got it all fixed, and ready for review

**Pull request merged**